### PR TITLE
feat: add `DeepReadonly` type

### PR DESCRIPTION
**Pull request by kyranet**

### CODE REVIEW

**Review by vladfrangu — DISMISSED** [review]

File: packages/iterator-utilities/package.json [hunk 1]
```diff
@@ -687,7 +687,7 @@
 		}
 	},
 	"sideEffects": [
-		"./dist/esm/chunk-G5GHKT7C.mjs",
```
Comment (favna) [inline packages/iterator-utilities/package.json:690]: Undo this change so it doesnt show up in the conventional commit scan as a feat for iterator-utilities

**Review by favna — CHANGES_REQUESTED** [review]

File: packages/iterator-utilities/package.json (hunk 1 above)
Comment (kyranet) [reply packages/iterator-utilities/package.json]: Done

**Review by favna — APPROVED** [review]

File: packages/utilities/README.md [hunk 2]
```diff
@@ -589,6 +590,21 @@ A union of all builtin types.
...
+
+```ts
+type Foo = Set<{ bar?: ['foo', { hello: 'world' }] }>;
+
+// ReadonlySet<{
+//     readonly bar?: readonly ["foo", {
+//         readonly hello: "world";
+//     }] | undefined;
+// }>
+declare const foo: DeepRequired<Foo>;
```
Comment (HigherOrderLogic) [inline packages/utilities/README.md:605]: This should be `DeepReadonly` right?

Comment (kyranet) [reply packages/utilities/README.md:605]: ... yes, nice catch, thank you!

Comment (kyranet) [reply packages/utilities/README.md:605]: Opened #776, thanks!
